XRTK-Core/Packages/com.xrtk.core/Extensions/GameObjectExtensions.cs

```diff
@@ -230,5 +230,23 @@ public static bool IsParentOrChildOf(this GameObject g1, GameObject g2)
 
             return t1.IsParentOrChildOf(t2);
         }
+
+        /// <summary>
+        /// Gets a component on the game object if it's already attached to it.
+        /// If the component does not exist on the game object, it will be attached and returned.
+        /// </summary>
+        /// <typeparam name="T">The type of the component to lookup on the game object.</typeparam>
+        /// <param name="g">GameObject instance.</param>
+        /// <returns>Existing instance of component or newly created one.</returns>
+        public static T GetOrAddComponent<T>(this GameObject g) where T : Component
```

Comment:
Let's pull this out into its own PR

Reply:
Agreed, general improvements not related to hands should go in their own PR for clarity